Assets/Scripts/PlayerController.cs

```diff
@@ -228,6 +237,17 @@ public class PlayerController : MonoBehaviour {
     }
 
     // Helper Methods
+    private void ResetPlayerPosition() {
+        rb.velocity = Vector3.zero;
+        rb.angularVelocity = Vector3.zero;
+        if (this.playerLastPlatform != null) {
+            // If player jumped off a platform but didn't hit any other ground, it should reset to the platform
+            var platformPosition = this.playerLastPlatform.transform.position;
+            playerLastPosition = platformPosition + new Vector3(0, (this.playerLastPlatform.transform.lossyScale.y / 2), 0);
```

Comment:
I see, the problem is that you cannot have a fixed last position when you want to be restored to on top of a moving object. Tricky. It would be nice though, if you could use the same solution for moving and non-moving objects. Like storing object + offset at all times.